crates/routes/src/utils/scheduled_tasks.rs

```diff
@@ -256,24 +256,36 @@ async fn process_post_aggregates_ranks_in_batches(conn: &mut AsyncPgConnection)
   while let Some(previous_batch_last_published) = previous_batch_result {
     let updated_rows = sql_query(
       r#"WITH batch AS (SELECT pa.post_id
-               FROM post_aggregates pa
-               WHERE pa.published > $1
-               AND (pa.hot_rank != 0 OR pa.hot_rank_active != 0)
-               ORDER BY pa.published
-               LIMIT $2
-               FOR UPDATE SKIP LOCKED)
-         UPDATE post_aggregates pa
-           SET hot_rank = r.hot_rank(pa.score, pa.published),
-           hot_rank_active = r.hot_rank(pa.score, pa.newest_comment_time_necro),
-           scaled_rank = r.scaled_rank(pa.score, pa.published, ca.users_active_month)
-         FROM batch, community_aggregates ca
-         WHERE pa.post_id = batch.post_id and pa.community_id = ca.community_id RETURNING pa.published;
-    "#,
+           FROM post_aggregates pa
+           WHERE pa.published > $1
+           AND (pa.hot_rank != 0 OR pa.hot_rank_active != 0)
+           ORDER BY pa.published
+           LIMIT $2
+          FOR UPDATE SKIP LOCKED),
+      community_interactions AS (
+          SELECT community_id, interactions_month
+          FROM community_aggregates
+          GROUP BY community_id)
+      UPDATE post_aggregates pa
+      SET hot_rank = r.hot_rank(pa.score, pa.published),
+          hot_rank_active = r.hot_rank(pa.score, pa.newest_comment_time_necro),
+          scaled_rank = r.scaled_rank(pa.score, pa.published, ci.interactions_month)
+      FROM batch, community_interactions ci
+      WHERE pa.post_id = batch.post_id
+      AND pa.community_id = ci.community_id
```

Comment:
the community_interactions subquery causes the whole community_aggregates table to be loaded every time, so do this instead, which is more similar to how it was before:

```suggestion
          FOR UPDATE SKIP LOCKED)
      UPDATE post_aggregates pa
      SET hot_rank = r.hot_rank(pa.score, pa.published),
          hot_rank_active = r.hot_rank(pa.score, pa.newest_comment_time_necro),
          scaled_rank = r.scaled_rank(pa.score, pa.published, ca.interactions_month)
      FROM batch, community_aggregates ca
      WHERE pa.post_id = batch.post_id
      AND pa.community_id = ca.community_id
```